### ISSUE TITLE
Sync: Failed to sync extension enablement

### ISSUE THREAD

**Issue by joaomoreno** [microsoft/vscode#91410]
Testing #91226 on Windows

I reprod this twice and then never again:

1. Machine A: disable a theme
2. Machine B: wait for it to be disabled due to sync, then re-enable it

Machine A would not see it re-enabled, no matter how long I waited.

Sorry for the poor steps... I was able to repro this with normal extensions (ESLint) as well as built-in (Monokai theme).

**Comment by sandy081** [comment]
@joaomoreno Can you please provide the sync log from both machines while this happening ?

**Comment by roblourens** [comment]
I also reproed this multiple times. I repro it often if I change multiple extensions enablement at the same time. Here I started with Gitlens and Java disabled in both windows, then enabled both in window A, but in window B, only Gitlens is enabled.

Window A:

```
[2020-02-25 16:05:34.660] [userDataSync] [info] Auto Sync: Triggered.
[2020-02-25 16:05:34.725] [userDataSync] [info] Settings: No changes found during synchronizing settings.
[2020-02-25 16:05:34.726] [userDataSync] [info] Keybindings: No changes found during synchronizing keybindings.
[2020-02-25 16:05:34.727] [userDataSync] [info] UI State: No changes found during synchronizing ui state.
[2020-02-25 16:05:34.949] [userDataSync] [info] Extensions: Updated remote extensions
[2020-02-25 16:05:34.968] [userDataSync] [info] Extensions: Updated last synchronized extensions
[2020-02-25 16:05:34.969] [userDataSync] [info] Sync done. Took 309ms
```

![image](https://user-images.githubusercontent.com/323878/75298983-bb9ed800-57e8-11ea-8c88-c3d0ee2bf0ad.png)

Window B:

```
[2020-02-25 16:05:36.399] [userDataSync] [info] Auto Sync: Triggered.
[2020-02-25 16:05:36.463] [userDataSync] [info] Settings: No changes found during synchronizing settings.
[2020-02-25 16:05:36.468] [userDataSync] [info] Keybindings: No changes found during synchronizing keybindings.
[2020-02-25 16:05:36.470] [userDataSync] [info] UI State: No changes found during synchronizing ui state.
[2020-02-25 16:05:36.799] [userDataSync] [info] Extensions: Enabled extension eamodio.gitlens 10.2.1
[2020-02-25 16:05:36.817] [userDataSync] [info] Extensions: Updated last synchronized extensions
[2020-02-25 16:05:36.817] [userDataSync] [info] Sync done. Took 418ms
[2020-02-25 16:05:38.805] [userDataSync] [info] Auto Sync: Triggered.
[2020-02-25 16:05:38.857] [userDataSync] [info] Settings: No changes found during synchronizing settings.
[2020-02-25 16:05:38.858] [userDataSync] [info] Keybindings: No changes found during synchronizing keybindings.
[2020-02-25 16:05:38.859] [userDataSync] [info] UI State: No changes found during synchronizing ui state.
[2020-02-25 16:05:38.891] [userDataSync] [info] Extensions: No changes found during synchronizing extensions.
[2020-02-25 16:05:38.891] [userDataSync] [info] Sync done. Took 86ms
```

![image](https://user-images.githubusercontent.com/323878/75299015-cce7e480-57e8-11ea-8e1b-e8d2b1400315.png)

**Comment by sandy081** [comment]
Assigning to feb for investigation

**Comment by sandy081** [comment]
@bpasero here is the code that is not sending storage change event when a key is removed from storage

https://github.com/microsoft/vscode/blob/fed69adea3fccae29fb04f6a0144da4acad9f8f0/src/vs/platform/storage/node/storageIpc.ts#L105-L107

Commit af9dddd0fcd65f310cc795e0c7c5370b4ca17086 introduced this.

**Comment by sandy081** [comment]
This should be fixed by #91582
